pom.xml

```diff
@@ -179,7 +171,17 @@
             <version>${mockito-junit-jupiter.version}</version>
             <scope>test</scope>
         </dependency>
-
+        <dependency>
```

Comment:
I don't think you should add a dependency on httpclient for this change. Yes, it has some nice functions for manipulating a URL but it's a very large and likely-conflicting-with-what-people-have dependency that ends up being overkill for this.